### ISSUE TITLE
CPU melting behavior

### ISSUE THREAD

**Issue by berkus** [spacedriveapp/spacedrive#2069]
### Describe the bug

While indexing the filesystem

<img width="291" alt="image" src="https://github.com/spacedriveapp/spacedrive/assets/80306/92c4fa3c-b37a-45b5-8d20-2c3d84b738f9">

So far it has peaked around 970% which is way more than cores I have and the macOS UI has noticeable lag with that.

Indexing fast sounds like a good idea, but leave some space for other apps too? Indexing should be background, not "I take it all".

### Reproduction

1. Download
2. Start
3. Add all recommended folders to index
4. Wait a little

### Expected behavior

App crawls files in the background without melting CPU and causing full-speed coolers on an M1 Pro (this is the first app that managed to do this, congrats!)

### Platform and versions

```Shell
macOS Sonoma 14.0

14-inch 2021 M1 Pro MacBook with 32GB RAM
```

**Comment by berkus** [comment]
<img width="283" alt="image" src="https://github.com/spacedriveapp/spacedrive/assets/80306/9075d9ec-8d03-4932-994e-f30a28c0ee6d">

¯\_(ツ)_/¯

**Comment by berkus** [comment]
<img width="269" alt="image" src="https://github.com/spacedriveapp/spacedrive/assets/80306/d48583f0-ce85-4eff-b16e-8762fdf089df">

So laggy I cannot take a screenshot ¯\_(ツ)_/¯

**Comment by utkubakir** [comment]
Which version are you on? We have setting (with default under 75% or 50%) that controls the cpu usage, can you also tell me the value of that setting?

Thanks.

**Comment by berkus** [comment]
That must be the latest, just installed it yesterday.

Version 0.2.3 (20240203.000617)

Thumbnailer CPU usage: 50%

**Comment by utkubakir** [comment]
I believe the new AI labeller we added might be the cause. Could you please test it on a folder without any images? It is unlikely that the indexer alone would render everything else unusable. 

Also, we are currently working on a new task system, set to be released in the next **alpha** update, which aims to address all these threading issues. (@fogodev)

**Comment by jamiepine** [comment]
Yeah that has to be the labeler, which likely doesn't have a throttle yet. Let's change this to a universal CPU setting and we can add individual ones down the line.

**Comment by utkubakir** [comment]
Fixed temporarily in `0.2.4` and will be fixed for good in the next version
